.github/workflows/pr-sanity.yaml

```diff
@@ -0,0 +1,27 @@
+name: Sanity Checks
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+  workflow_dispatch:
+
+jobs:
+  build:
+    name: Sanity Checks
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - uses: actions/setup-go@v2
+        with:
+         go-version: '1.13'
+
+      - name: Update metrics documentation
+        run: make generate-doc
+
+      - name: Validate there are no pending changes
+        run: git difftool -y --trust-exit-code --extcmd=./hack/diff-csv.sh
```

Comment:
maybe we can use something like `[ -n "$(git status --porcelain)" ] && exit 1` instead the `/diff-csv.sh` script ?

Reply:
using `git difftool` allows users to see in the logs what were the diffs that caused the test to fail